src/rules/date-rules.ts

```diff
@@ -15,6 +15,7 @@ export const DateRuleHelpers = {
   isBetweenDates (minDate: Date, maxDate: Date, dateStrToValidate: string): boolean {
     if (!dateStrToValidate) return true
     const date = new Date(dateStrToValidate) // eg, September 5, 2022
+    minDate.setHours(0, 0, 0) // Removing the hour to properly compare in case selected date was same as minimum.
     return (date >= minDate) && (date <= maxDate)
```

Comment:
There was a bit of weird behavior here. When selecting the date as same as registration date, the error (rule) was being shown:
![rule is shown](https://github.com/bcgov/business-create-ui/assets/122301442/3f7d2468-a3b8-4375-bcdd-fdf62d5cb574)

Apparently, the issue is with the hour:
![weird dates](https://github.com/bcgov/business-create-ui/assets/122301442/138257dc-7463-473a-aded-22cebb35c3bb)
The first line is the `date` while the second one is the `minDate`. Hence, the rule is being triggered.

Setting the hour to be the same (`00:00:00`) fixes this issue.

Reply:
I think this is indeed the root cause 👍 

What do you think about fixing maxDate similarly (and also isNotBeforeDate() and isNotAfterDate()) ?